daemon/src/rollover/protocol.rs

```diff
@@ -141,3 +301,302 @@ pub(crate) async fn emit_failed(order_id: OrderId, e: anyhow::Error, executor: &
+            (counterparty_pk, other_punish_params),
+        ),
+        Role::Taker => (
+            (counterparty_pk, other_punish_params),
+            (pk, own_punish_params),
+        ),
+    };
+
+    (punish_maker, punish_taker, other_punish_params)
+}
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn build_own_cfd_transactions(
+    dlc: &Dlc,
+    rollover_params: RolloverParams,
+    announcement: &olivia::Announcement,
+    oracle_pk: schnorrsig::PublicKey,
+    our_role: Role,
+    our_position: Position,
+    n_payouts: usize,
+    complete_fee: FeeFlow,
+    ((maker_identity, maker_punish_params), (taker_identity, taker_punish_params)): (
+        (PublicKey, PunishParams),
+        (PublicKey, PunishParams),
+    ),
+) -> Result<CfdTransactions> {
+    let sk = dlc.identity;
+
+    let maker_lock_amount = dlc.maker_lock_amount;
+    let taker_lock_amount = dlc.taker_lock_amount;
+    let payouts = HashMap::from_iter([(
+        Announcement {
+            id: announcement.id.to_string(),
+            nonce_pks: announcement.nonce_pks.clone(),
+        },
+        calculate_payouts(
+            our_position,
+            our_role,
+            rollover_params.price,
+            rollover_params.quantity,
+            rollover_params.long_leverage,
+            rollover_params.short_leverage,
+            n_payouts,
+            complete_fee,
+        )?,
+    )]);
+
+    // unsign lock tx because PartiallySignedTransaction needs an unsigned tx
+    let mut unsigned_lock_tx = dlc.lock.0.clone();
+    unsigned_lock_tx
+        .input
+        .iter_mut()
+        .for_each(|input| input.witness.clear());
+
+    let lock_tx = PartiallySignedTransaction::from_unsigned_tx(unsigned_lock_tx)?;
+    let own_cfd_txs = tokio::task::spawn_blocking({
+        let maker_address = dlc.maker_address.clone();
+        let taker_address = dlc.taker_address.clone();
+        let lock_tx = lock_tx.clone();
+
+        move || {
+            renew_cfd_transactions(
+                lock_tx,
+                (
+                    maker_identity,
+                    maker_lock_amount,
+                    maker_address,
+                    maker_punish_params,
+                ),
+                (
+                    taker_identity,
+                    taker_lock_amount,
+                    taker_address,
+                    taker_punish_params,
+                ),
+                oracle_pk,
+                (CET_TIMELOCK, rollover_params.refund_timelock),
+                payouts,
+                sk,
+                rollover_params.fee_rate.to_u32(),
+            )
+        }
+    })
+    .await?
+    .context("Failed to create new CFD transactions")?;
+
+    Ok(own_cfd_txs)
+}
+
+pub(crate) fn build_commit_descriptor(
+    ((maker_identity, maker_punish_params), (taker_identity, taker_punish_params)): (
+        (PublicKey, PunishParams),
+        (PublicKey, PunishParams),
+    ),
+) -> Descriptor<PublicKey> {
```

Comment:
This doesn't have to be a tuple of tuples. You can get rid of at least one level